code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -1270,7 +1270,7 @@ var/global/list/damage_icon_parts = list()
 		var/icon/tailw_s = new/icon("icon" = 'icons/effects/species.dmi', "icon_state" = "[tail]w_s")
 		if(species.bodyflags & HAS_SKIN_COLOR)
 			tailw_s.Blend(rgb(r_skin, g_skin, b_skin), ICON_ADD)
-		if(tail_marking_icon)
+		if(tail_marking_icon && !tail_marking_style.tails_allowed)

```

Comment:
Because this is in 'tail' code not 'body accessory' code, it should be perfectly fine because the only things in a marking style's tails_allowed attribute are body accessories. All the tail markings for non body-accessory tails have empty/null tails_allowed, which is what this code is for and thus is why I did it this way